package_scanner.nim

```diff
@@ -1,135 +1,367 @@
+## Package JSON Linter
+## ===================
+##
+## - Tests the ``package.json`` list from this repository.
+##
+## Check the packages for:
+## - Validate by parsing with NodeJS
+## - Validate by parsing with Python
+## - Validate by parsing with Ruby
+## - Missing name
+## - Missing/unknown method
+## - Missing/unreachable repository
+## - Missing tags
+## - Missing description
+## - Missing/unknown license
+## - Insecure git:// url on GitHub
+## - Insecure http:// url
+## - Blacklisted tags
+## - Empty string tags
+## - Unwanted whitespaces on tags
+## - Too much tags
+## - Missing nimble files on repos
+## - Packages with duplicated URLs
+## - Enforce required keys
+## - Maximum number of tags
+## - Maximum lenght of tags
+## - Alias relate to Names
+## - Profanity Filter by Levenshtein Distance.
+## - Try to download the nimble file from the repo.
+## - Try to Git clone the repo (deletes folder after).
+## - Works online/offline
+##
+## Use
+## ---
+##
+## - Off-line: ``nim c -r package_scanner2.nim`` (Faster, wont use Internet).
```

Comment:
I'd rather this be an explicit option, silently disabling internet checks could cause problems down the road.

Reply:
Its not Silent, it prints a Hint using the Hint Pragma with explicit message.
Note that by default it runs with with SSL then it goes online.
But I get the idea, you want another additional `define` :grey_question:  :thinking:

Reply:
I would prefer a run-time argument. Also I want the default behaviour to be "online".

Reply:
The default behavior to be "online".
No extra option required anywhere.
All other is for when a dev want to run it manually for whatever reason.

:slightly_smiling_face:

Reply:
No, the default is ``offline``. Default means "compiling with no options".